cluster-provision/images/vm-image-builder/customize-image.sh

```diff
@@ -27,6 +35,22 @@ cloud-localds "${CLOUD_INIT_ISO}" "${CLOUD_CONFIG_PATH}"
 echo "Customize image by booting a VM with
  the image and cloud-init disk
  press ctrl+] to exit"
+
+# Check if it is native build, if true use kvm
+# otherwise use emulation
+if [[ ${ARCH} = $(uname -m) ]]; then
+  buildconfig="--virt-type kvm"
+else
+  buildconfig="--arch ${ARCH}"
```

Comment:
AFAICT this is the only spot in the entire thing where we actually need Linux-style architecture names, because virt-install expects those. We should do a conversion on the spot for this specific purpose, and use Go-style architecture names everywhere else.